### DISCUSSION TITLE
Permit more Devise authentication keys

### DISCUSSION THREAD

**Discussion by armstnp** [graphql-devise/graphql_devise#139] [General]
Currently, the gem seems to have e-mail hard-wired as the only authentication key. Would it be tenable to parameterize the mutations based on the application's configured Devise authentication keys?

**Comment by mcelicalderon** [comment]
Hey @armstnp! Sorry for the late reply. I think this would be nice. We just need to take a closer look as we still depend a lot on DTA's implementation and I'm almost certain that they have email hardwired too for email authentication (they also support oauth). A change like this might drive us further away from depending on DTA, but that's something we are willing to do in the future in order to gain more flexibility on what this gem can and cannot do.
